### PR TITLE
Loader: port or remove selftests

### PR DESCRIPTION
**Pull request by clebergnu**
With the upcoming removal of the loader (and legacy runner), let's salvage what is important.

Some of the loader selftests have been covering functionality not covered by the resolver, so they need to be ported.  On the other hand, there are duplicate tests that can simply be removed.

### CODE REVIEW

File: selftests/unit/plugin/test_resolver.py [hunk 1]
```diff
@@ -113,16 +113,15 @@ def test_exec_test(self):
...
-
+        self.assertEqual(res.reference, exec_test.path)
+        self.assertEqual(res.result, resolver.ReferenceResolutionResult.SUCCESS)
+        self.assertEqual(len(res.resolutions), 1)
+        resolution = res.resolutions[0]
+        self.assertEqual(resolution.kind, 'exec-test')
+        self.assertEqual(resolution.uri, exec_test.path)
+        self.assertEqual(resolution.args, ())
+        self.assertEqual(resolution.kwargs, {})
+        self.assertEqual(resolution.tags, None)
```
Comment (willianrampazzo) [inline selftests/unit/plugin/test_resolver.py:124]: I don't know why, but I have these lines showing they were changed, but I don't see any change. This is the case for this commit.

Comment (clebergnu) [reply selftests/unit/plugin/test_resolver.py:124]: They were just re-indented... not sure if that's the reason.

Comment (willianrampazzo) [reply selftests/unit/plugin/test_resolver.py:124]: Ah, interesting, it is not even showing the indentation here.

**Review by willianrampazzo — APPROVED** [review]
+25 -356 \o/

**Review by beraldoleal — APPROVED** [review]

**Pull request merged**